Automattic/jetpack
# Repeat Visitor: register new block and assets

#### Changes proposed in this Pull Request:
This change adds a new "Repeat Visitor" block that controls its Inner Block visibility based on how often a visitor has viewed the page. It can be configured to show its content either above or below a certain view count threshold.

The corresponding wp-calypso PR is [here](https://github.com/Automattic/wp-calypso/pull/30484).

#### Testing instructions:





1. Spin up a [Jurassic Ninja instance](https://jurassic.ninja/create/?gutenpack&calypsobranch=try/visited-block&branch=try/visited-block).

2. Connect Jetpack to your WPCOM account.

3. Navigate to Settings->Jetpack Constants, check JETPACK_BETA_BLOCKS, and Save.

4. Add the Visited block to a post or page. Nest whatever block within the Visited block.

5. Save a post and open it. 

6. Check the cookie value for `jp-visit-counter`; it should have incremented from `0` to `1`. 

7. Refresh the page multiple times and ensure that the `jp-visit-counter` continues to increment. Ensure that your content becomes visible after it crosses the minimum view threshold (and vice versa for maximum view threshold).

#### Proposed changelog entry for your changes:

* Adds a Repeat Visitor block that controls block visibility based on how often a visitor has viewed the page.
